### PR TITLE
Add initial functionality for I2C and SPI.

### PR DESCRIPTION
**Pull request by caternuson**
* basic readings: pressure, temperature, altitude
* over sampling individually settable for pressure and temperature
* IIR filter coefficient

### CODE REVIEW

File: adafruit_bmp3xx.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+class BMP3XX:
+    """Base class for BMP3XX sensor."""
+
+    def __init__(self):
+        chip_id = self._read_byte(_REGISTER_CHIPID)
+        if _CHIP_ID != chip_id:
+            raise RuntimeError('Failed to find BMP3XX! Chip ID 0x%x' % chip_id)
+        self._read_coefficients()
+        self.reset()
+        self.sea_level_pressure = 1013.25 # hPa @ MSL
```
Comment (tannewt) [inline adafruit_bmp3xx.py]: Add a """comment""" after this so sea_level_pressure is shown in RTD.

**Review by tannewt — CHANGES_REQUESTED** [review]
One thing but good otherwise.

**Comment by ladyada** [comment]
i tested this code, seems to work fine so after the minor update, we can merge!

**Review by ladyada — APPROVED** [review]
tested with hardware

**Review by tannewt — APPROVED** [review]
Ok, looks good! Thanks!

**Pull request merged**